### PR TITLE
add warning about validation_split

### PR DESCRIPTION
**Pull request by pwiercinski**
### Summary
Adding a warning describing the way validation data is selected when `model.fit` is run with `validation_split`.
The fact that `validation_split` simply selects the last examples from the data set is not obvious and displaying this warning would surely save plenty of time anyone who may assume otherwise as well as serve as a reminder to those who might not have considered it an issue.

### PR Overview

- [x] This PR requires new unit tests [n] (make sure tests are included)
- [x] This PR requires to update the documentation [n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y]
- [x] This PR changes the current API [n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
We can't really display a part of the documentation as Python warnings. The documentation and the python warnings have two very different purposes. We expect users to read the documentation at least once before starting working.

> The fact that validation_split simply selects the last examples from the data set is not obvious

Maybe not obvious, but it's the expected behavior and this behavior is desirable in 99% of use cases, it allows users to do the right thing without thinking about it. Splitting after shuffle is a pretty dangerous thing to do, you have the risk of using data from the target domain into you train set. If your data was collected at different point in time, it would mean that you can be training on future data and predicting on past data.

Do you have a use case where one would want shuffling before splitting?

**Comment by pwiercinski** [comment]
Thanks for reviewing.

> Do you have a use case where one would want shuffling before splitting?

Sure, in any classification task. In fact I don't think there is a use case where one would want to shuffle the training set, but not the validation set - which is what keras does by default (`shuffle=True` by default). This indeed is the source of confusion. Do you have a use case where one would want to shuffle the training data set and not the validation data set?

> Splitting after shuffle is a pretty dangerous thing to do, you have the risk of using data from the target domain into you train set. If your data was collected at different point in time, it would mean that you can be training on future data and predicting on past data.

I agree this is the case in time series problems, but then again in such case one would not want to shuffle the training set either. In any other case where specific data for validation set is required surely the `validation_data` is the appropriate flag.
